src/towncrier/_project.py

```diff
@@ -76,3 +77,5 @@ def get_project_name(package_dir, package):
     if isinstance(version, Version):
         # Incremental has support for package names
         return version.package
+
+    raise TypeError(f"Unsupported type for __version__: {type(version)}")
```

Comment:
The previous code would implicitly return `None` here, but the caller couldn't handler `None`, so I believe that was a bug.